octoprint_mrbeam/wizard_config.py

```diff
@@ -0,0 +1,149 @@
+from flask.ext.babel import gettext
+from octoprint_mrbeam.mrb_logger import mrb_logger
+
+
+class WizardConfig:
+	def __init__(self, plugin):
+		self._logger = mrb_logger("octoprint.plugins.mrbeam.wizard_config")
+
+		self._plugin = plugin
+		self._user_manager = plugin._user_manager
+		self._plugin_manager = plugin._plugin_manager
+		self._settings = plugin._settings
+
+		self._wizard_version = 16  # random number. but we can't go down anymore, just up.
+
+		self._is_welcome_wizard = plugin.isFirstRun()
+		self._is_whatsnew_wizard = not plugin.isFirstRun()
+
+		self._current_wizard_config = None
+
+	def get_wizard_version(self):
+		return self._wizard_version
+
+	def get_wizard_config_to_show(self):
+		wizard_config_to_show = []
+
+		if self._is_welcome_wizard:
+			self._current_wizard_config = self._welcome_wizard_config()
+		elif self._is_whatsnew_wizard:
+			self._current_wizard_config = self._whatsnew_wizard_config()
+
+		for wizard, config in self._current_wizard_config.iteritems():
+			required = config.get('required', False)
+			if required:
+				config.pop('required', None)
+				wizard_config_to_show.append(config)
+
+		return wizard_config_to_show
+
+	def _welcome_wizard_config(self):
+		"""Add here the tabs that should be present in the welcome wizard"""
+		welcome_wizard_tabs = dict(
+			wizard_lasersafety=dict(
+				type='wizard',
+				name=gettext("Laser Safety"),
+				required=True,
+				mandatory=False,
+				suffix='_lasersafety',
+				template='wizard/wizard_lasersafety.jinja2',
+				div='wizard_plugin_corewizard_lasersafety',
+			),
+			wizard_wifi=dict(
+				type='wizard',
+				name=gettext("Wifi Setup"),
+				required=self._is_wifi_wizard_required(),
+				mandatory=False,
+				suffix='_wifi',
+				template='wizard/wizard_lasersafety.jinja2',
+				div='wizard_plugin_corewizard_lasersafety',
+			),
+			wizard_acl=dict(
+				type='wizard',
+				name=gettext("Access Control"),
+				required=self._is_acl_wizard_required(),
+				mandatory=False,
+				suffix='_acl',
+				template='wizard/wizard_acl.jinja2',
+				div='wizard_plugin_corewizard_acl',
+			),
+			wizard_analytics=dict(
+				type='wizard',
+				name=gettext("Analytics"),
+				required=self._is_analytics_wizard_required(),
+				mandatory=False,
+				suffix='_analytics',
+				template='wizard/wizard_analytics.jinja2',
+				div='wizard_plugin_corewizard_analytics',
+			)
+		)
+
+		return welcome_wizard_tabs
+
+	@staticmethod
```

Comment:
Why is this one a static method while _welcome_wizard_config() is a regular member?
Does the call to this one even work with self._whatsnew_wizard_config() (line 30)

Reply:
Changed